flytekit/extras/sqlalchemy/task.py

```diff
@@ -0,0 +1,84 @@
+import typing
+from dataclasses import dataclass
+
+import pandas as pd
+from sqlalchemy import create_engine
+
+from flytekit import current_context, kwtypes
+from flytekit.core.base_sql_task import SQLTask
+from flytekit.core.python_function_task import PythonInstanceTask
+from flytekit.types.schema import FlyteSchema
+
+
+@dataclass
+class SQLAlchemyConfig(object):
+    """
+    Use this configuration to configure task. String should be standard
+    sqlalchemy connector format
+    (https://docs.sqlalchemy.org/en/14/core/engines.html#database-urls).
+    Database can be found:
+      - within the container
+      - or from a publicly accessible source
+
+    Args:
+        uri: default sqlalchemy connector
+        connect_args: sqlalchemy kwarg overrides -- ex: host
+        secret_connect_args: flyte secrets loaded into sqlalchemy connect args
+            -- ex: {"password": {"name": SECRET_NAME, "group": SECRET_GROUP}}
+    """
+
+    uri: str
+    connect_args: typing.Optional[typing.Dict[str, typing.Any]] = None
+    secret_connect_args: typing.Optional[typing.Dict[str, typing.Dict[str, typing.Any]]] = None
```

Comment:
why not another dataclass? Its always 2 values at max right? Actual you can just use the secrets class here - https://github.com/flyteorg/flytekit/blob/8528268a29a07fe7e9ce9f7f08fea68c41b6a60b/flytekit/models/security.py#L11 right?